hbase-http/src/main/java/org/apache/hadoop/hbase/http/ServletConfig.java

```diff
@@ -23,14 +23,20 @@
 
 @InterfaceAudience.Private
 class ServletConfig {
-  private String name;
-  private String pathSpec;
-  private String clazz;
+  private final String name;
+  private final String pathSpec;
+  private final String clazz;
+  private final boolean expected;
 
   public ServletConfig(String name, String pathSpec, String clazz) {
+    this(name, pathSpec, clazz, true);
```

Comment:
The default expected is true? Then how could this reduce the noisy log messages?

Reply:
MetricsServlet is the only class which isn't bundled in hbase itself. The default is true because if those other metric classes fail, that does seem odd and we should warn. We pass `false` for MetricsServlet because it's not expected to be on the classpath for most users, so should not warn.

If a different naming/organization would make this more obvious, I'm happy to make a change.